distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1985,40 +1770,48 @@ public void testCancelDownload() throws Exception {
                 RESOURCE_DOWNLOAD_ENDPOINT_ROOT.getUrl() + "?source=" + CSW_STUB_SOURCE_ID
                         + "&metacard=" + metacardId;
 
-        // @formatter:off
-        String downloadId = given().auth().preemptive().basic(LOCALHOST_USERNAME, LOCALHOST_PASSWORD)
-                .get(startDownloadUrl).then().log().all()
-                .extract().jsonPath().getString("downloadId");
-        // @formatter:on
+        given().auth()
+                .preemptive()
+                .basic(LOCALHOST_USERNAME, LOCALHOST_PASSWORD)
+                .get(startDownloadUrl)
+                .then()
+                .log()
+                .all();
 
-        localhostCometDClient.cancelDownload(downloadId);
+        expect("Waiting for download to start.").within(30, SECONDS)
+                .until(() -> localhostCometDClient.getDownloadIds()
+                        .size() > 0);
 
-        expect("Waiting for notifications.").within(10, SECONDS)
-                .until(() -> localhostCometDClient.getMessages(NOTIFICATIONS_CHANNEL)
-                        .size() == 1);
+        assertThat(localhostCometDClient.getDownloadIds()
+                .size(), is(1));
 
-        /**
-         * Wait for 2 activities. The first one is the download started, and the second one is the
-         * download canceled.
-         */
-        expect("Waiting for activities.").within(10, SECONDS)
-                .until(() -> localhostCometDClient.getMessages(ACTIVITIES_CHANNEL)
-                        .size() == 2);
+        String downloadId = localhostCometDClient.getDownloadIds()
```

Comment:
Similarly, couldn't the download have completed by here, resulting in a `NoSuchElementException`?